mpqp/noise/noise_model.py

```diff
@@ -321,7 +323,116 @@ def to_kraus_representation(self) -> KrausRepresentation: ...
 
 
 class AmplitudeDamping(NoiseModel):
-    """3M-TODO"""
+    """Class representing the amplitude damping noise channel, which can model both
+    the standard and generalized amplitude damping processes. It can be applied
+    to a single qubit and depends on two parameters: the decay rate `gamma` and the
+    probability of excitation `prob`.
+
+    Args:
+        gamma: Decaying rate of the amplitude damping noise channel.
+        prob: Probability of excitation in the generalized amplitude damping noise channel
+            When set to 0, indicating standard amplitude damping.
+        targets: List of qubit indices affected by this noise.
+        gates: List of :class:`Gates<mpqp.core.instruction.gates.gate.Gate>`
+            affected by this noise.
+
+    Raises:
+        ValueError: When the gamma or prob parameters are outside of the expected interval [0, 1].
+        ValueError: When no target qubits are specified.
+
+    Examples:
+        >>> circuit = QCircuit([H(i) for i in range(3)])
+        >>> ad1 = AmplitudeDamping(0.2, 0, [0])
+        >>> ad2 = AmplitudeDamping(0.4, 0.1, [1, 2])
+        >>> circuit.add([ad1, ad2])
+        >>> print(circuit)
+             ┌───┐
+        q_0: ┤ H ├
+             ├───┤
+        q_1: ┤ H ├
+             ├───┤
+        q_2: ┤ H ├
+             └───┘
+        NoiseModel:
+            AmplitudeDamping(gamma=0.2, prob=0, targets=[0])
+            AmplitudeDamping(gamma=0.4, prob=0.1, targets=[1, 2])
+
+    """
+
+    def __init__(
+        self,
+        gamma: float,
+        prob: float,
```

Comment:
We should put to optional and the default value to 1, so if the user wants the basic amplitude damping, he only inputs gamma, no need to touch prob